src/main/java/cz/muni/fi/pv286/mmt/model/ResultTree.java

```diff
@@ -0,0 +1,58 @@
+package cz.muni.fi.pv286.mmt.model;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * ResultTree structure is used for array parser.
+ */
+public class ResultTree {
+    private byte[] value;
+    String intermediate;
+    private ArrayList<ResultTree> children = new ArrayList<>();
+    private int childIndex = 0;
+
+    public ResultTree(byte[] value) {
+        this.value = value;
+    }
+
+    public ResultTree(String intermediate) {
+        this.intermediate = intermediate;
+    }
+
+    public ResultTree(ArrayList<ResultTree> children) {
+        this.children = children;
+    }
+
+    public byte[] getValue() {
+        return value;
+    }
+
+    public void setValue(byte[] value) {
+        this.value = value;
+    }
+
+    public List<ResultTree> getChildren() {
+        return children;
+    }
+
+    public void setChildren(ArrayList<ResultTree> children) {
```

Comment:
**todo:** The type should be an interface.

The type of "children" should be an interface such as "List" rather than the implementation "ArrayList".